src/tests/class_tests/openms/source/MRMTransitionGroupPicker_test.cpp

```diff
@@ -454,6 +493,56 @@ START_SECTION((template < typename SpectrumT > void remove_overlapping_features(
 }
 END_SECTION
 
+START_SECTION(( void calculateBgEstimationAverage_(const MSChromatogram& chromatogram,
+  double best_left, double best_right, double & background, double & avg_noise_level) ))
+{
+
+  RichPeakChromatogram chromatogram;
+  setup_toy_chromatogram(chromatogram);
+
+  // Features
+  double best_left = 2.477966667;
+  double best_right = 3.01895;
+  double background, noise_level;
+
+  // Correct the background
+  MRMTransitionGroupPicker picker;
+
+  picker.calculateBgEstimationAverage_(chromatogram, 
+    best_left, best_right, background,
+    noise_level);
+
+  TEST_REAL_SIMILAR(background, 125076);
+  TEST_REAL_SIMILAR(noise_level, 2233.5);
+}
+END_SECTION
+
+START_SECTION(( void calculateBgEstimationExact_(const MSChromatogram& chromatogram,
+  double best_left, double best_right, double peak_height, double & background, double & avg_noise_level) ))
+{
+
+  RichPeakChromatogram chromatogram;
+  setup_toy_chromatogram(chromatogram);
+
+  // Features
+  double best_left = 2.477966667;
+  double best_right = 3.01895;
+  double peak_height = 966489;
+  double background, noise_level;
+
+  // Correct the background
+  MRMTransitionGroupPicker picker;
+
+  picker.calculateBgEstimationExact_(chromatogram, 
+    best_left, best_right, 
+    peak_height, background,
+    noise_level);
+
+  TEST_REAL_SIMILAR(background, 1140.392866);
+  TEST_REAL_SIMILAR(noise_level, 1908.596906);
```

Comment:
If you compare the two results, it seems also clear here that when you apply this in the code

```
           intensity_sum -= background;
           peak_apex_int -= avg_noise_level;
```

that with this background estimate, you are not getting the right / expected result